src/rtcModule/webrtc.cpp

```diff
@@ -2955,6 +2998,44 @@ void Session::onSignalingChange(webrtc::PeerConnectionInterface::SignalingState
 void Session::onDataChannel(webrtc::DataChannelInterface*)
 {}
 
+void Session::onTrack(rtc::scoped_refptr<webrtc::RtpTransceiverInterface> transceiver)
+{
+    SUB_LOG_DEBUG("onTrack:");
+    if (mState != kStateInProgress)
+    {
+        return;
+    }
+
+    if (transceiver->media_type() == cricket::MEDIA_TYPE_VIDEO)
+    {
+        mRemotePlayer->attachVideo(transceiver->receiver()->streams()[0]->GetVideoTracks()[0]);
+        mRemotePlayer->enableVideo(mPeerAv.video());
+    }
+    else if (transceiver->media_type() == cricket::MEDIA_TYPE_AUDIO)
+    {
+        mRemotePlayer->getAudioTrack()->AddSink(mAudioLevelMonitor.get());
+    }
+}
+
+void Session::onRenegotiationNeeded()
+{
+    if (mState != kStateInProgress)
+    {
+        return;
+    }
+
+    if (mRenegotiationInProgress)
+    {
+        SUB_LOG_WARNING("Ignoring multiple calls of onRenegotiationNeeded");
+        return;
+    }
+
+    mRenegotiationInProgress = true;
+    setStreamRenegotiationTimeout();
```

Comment:
I would move the `mRenegotiationInProgress = true;` into the method that actually sets the corresponding timer. I'll push a commit for that.